community/kernel/src/main/java/org/neo4j/unsafe/impl/batchimport/ReadNodeRecordsByCacheStep.java

```diff
@@ -80,7 +95,9 @@ public void run()
         @Override
         public void change( long nodeId, ByteArray array )
         {
-            batch[cursor++] = nodeStore.getRecord( nodeId, nodeStore.newRecord(), RecordLoad.CHECK );
+            boolean inUse = recordCursor.next( nodeId );
+            assert inUse;
```

Comment:
Would it make sense to acquire cursor in `#start()` with `RecordLoad.NORMAL` and skip this assertion?